tests/io/vasp/test_help.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+import io
+from unittest.mock import patch
+
+import pytest
+import requests
+
+from pymatgen.io.vasp.help import VaspDoc
+
+try:
+    from bs4 import BeautifulSoup
+except ImportError:
+    BeautifulSoup = None
+
+
+try:
+    website_down = requests.get("https://www.vasp.at", timeout=60).status_code != 200
+except requests.exceptions.ConnectionError:
+    website_down = True
+
+
+@pytest.mark.skipif(BeautifulSoup is None, reason="BeautifulSoup4 is not installed")
+@pytest.mark.skipif(website_down, reason="www.vasp.at is down")
+class TestVaspDoc:
+    @pytest.mark.parametrize("tag", ["ISYM"])
+    def test_print_help(self, tag):
+        vasp_doc = VaspDoc()
+        with patch("sys.stdout", new=io.StringIO()) as fake_stdout:
+            vasp_doc.print_help(tag)
+            output = fake_stdout.getvalue()
+
+        assert tag in output
+
+    @pytest.mark.parametrize("tag", ["ISYM"])
+    def test_get_help(self, tag):
+        vasp_doc = VaspDoc()
+        docstr = vasp_doc.get_help(tag)
+
+        assert tag in docstr
+
+    def test_get_incar_tags(self):
+        incar_tags = VaspDoc.get_incar_tags()
+        assert incar_tags
```

Comment:
can we assert `incar_tags` type and/or actual value?

Reply:
Good point actually. I didn't know what to `assert` in this first place because these INCAR tags are fetched from VASP wiki and might be prone to change (so if we assert some INCAR tags are in the list, they might be removed at some point, though sounds very unlikely to happen any time soon).

Reply:
Can I have a quick comment on the following @jansoh? Thanks in advance.

2ede975059dab8b68fdd4513516db9b7dcc3a590 seems to reveal quite some difference between INCAR tags from VASP wiki and `incar_parameters.json` file. **I'm not sure if VASP wiki is the complete set of INCAR tags, and what about VASP versioning (does VASP wiki only display INCAR tags for the latest version of VASP)?** 

Tags in json file but not in VASP wiki (remove them from json file) (**are they just absent from latest version of VASP?**):
```
'SHIFTRED', 'ML_FF_NR2_MB', 'ML_FF_MSPL2_MB', 'ML_FF_IREG_MB', 'ML_FF_MCONF_NEW', 'ML_FF_ISAMPLE', 'ML_FF_NHYP2_MB', 'ML_FF_NHYP1_MB', 'LCOMPAT', 'LSCALAPACK', 'ML_FF_NATOM_COUPLED_MB', 'ML_FF_IWEIGHT', 'ORBITALMAG', 'LSCSGRAD', 'LTETE', 'MAGDIPOLOUT', 'ML_FF_WTIFOR', 'ML_FF_MCONF', 'NDAV', 'ML_FF_RCUT2_MB', 'LVDWEXPANSION', 'ML_FF_LHEAT_MB', 'LSEPK', 'LVHAR', 'LSINGLES', 'LSEPB', 'ML_FF_LBASIS_DISCARD', 'MAGPOS', 'ML_FF_CSLOPE', 'LUSE_VDW', 'MINROT', 'ENCUTLF', 'LTRIPLET', 'ML_FF_LCOUPLE_MB', 'LUSEW', 'DARWINV', 'LORBITALREAL', 'LHFONE', 'LSUBROT', 'ML_FF_ICUT1_MB', 'ML_FF_NMDINT', 'ISPIND', 'ML_FF_WTOTEN', 'ML_FF_LCONF_DISCARD', 'LMETAGGA', 'ML_FF_CTIFOR', 'ML_FF_ICUT2_MB', 'LWAVE', 'LSELFENERGY', 'ML_FF_LNORM2_MB', 'LSPIRAL', 'LFXCEPS', 'MDALGO', 'LVEL', 'MAXMEM', 'ML_FF_SION1_MB', 'LWRITE_UNK', 'ML_FF_LMAX2_MB', 'ML_FF_EATOM', 'ENMAX', 'ML_FF_RCOUPLE_MB', 'ML_FF_SIGV0_MB', 'LMUSIC', 'ML_FF_AFILT2_MB', 'METAGGA', 'LRPAFORCE', 'LDOWNSAMPLE', 'ML_FF_CDOUB', 'ML_FF_W1_MB', 'ML_FF_LAFILT2_MB', 'ML_FF_LMLMB', 'ML_FF_ISTART', 'ML_FF_SION2_MB', 'NUCIND', 'ML_FF_ISOAP1_MB', 'SELFENERGY', 'LNICSALL', 'ML_FF_MRB2_MB', 'LWANNIER90', 'LTCTE', 'ML_FF_LNORM1_MB', 'ML_FF_ICOUPLE_MB', 'ML_FF_CSF', 'ML_FF_MHIS', 'ML_FF_CSIG', 'LFXHEG', 'LSYMGRAD', 'DARWINR', 'LFERMIGW', 'LSCALU', 'LSOL', 'ML_FF_NWRITE', 'ML_FF_IERR', 'LVDW_EWALD', 'LSORBIT', 'ML_FF_ISOAP2_MB', 'ML_FF_MB_MB', 'LREAL', 'ML_FF_LEATOM_MB', 'ML_FF_IBROAD2_MB', 'ML_FF_NR1_MB', 'STM', 'LVTOT', 'LSPECTRALGW', 'LGAUGE', 'ML_FF_W2_MB', 'ML_FF_NDIM_SCALAPACK', 'M_CONSTR', 'ML_FF_MSPL1_MB', 'HFSCREENC', 'LZEROZ', 'ML_FF_MRB1_MB', 'ML_FF_LMLFF', 'LRHFCALC', 'MIXFIRST', 'MIXPRE', 'MAXMIX', 'ML_FF_IBROAD1_MB', 'LRPA', 'LSCAAWARE', 'LWRITE_WANPROJ', 'LTHOMAS', 'ML_FF_ISCALE_TOTEN_MB', 'ML_FF_IAFILT2_MB', 'ML_FF_WTSIF', 'LWRITE_MMN_AMN', 'LMAGBLOCH', 'LSPECTRAL', 'ML_FF_LCRITERIA', 'LWANNIER90_RUN', 'ML_FF_RCUT1_MB', 'MAGMOM', 'ML_FF_SIGW0_MB'
```

Tags in VASP wiki file but not in json file (add them to json file):
```
'PHON_BORN_CHARGES', 'WANNIER90_WIN', 'LMP2LT', 'HFALPHA', 'BANDGAP', 'LPHON_DISPERSION', 'CMBJE', 'LIBMBD_TS_SR', 'LDISENTANGLED', 'ML_LBASIS_DISCARD', 'ALPHA_VDW', 'ML_WTOTEN', 'CPARAM', 'AMGGAX', 'IRC_MAXSTEP', 'LIBMBD_MBD_BETA', 'ML_MB', 'PHON_SIGMA', 'LIBMBD_K_GRID', 'ML_WTIFOR', 'SCISSOR', 'RSMBJ', 'IRC_VNORM0', 'ML_IWEIGHT', 'LKPOINTS_WAN', 'ML_NATOM_COUPLED', 'NATURALO', 'ML_RCUT2', 'ML_RDES_SPARSDES', 'CUTOFF_MU', 'ML_ICOUPLE', 'BPARAM', 'ML_SION2', 'ML_LEATOM', 'PHON_DOS', 'SCALEE', 'ML_ISCALE_TOTEN', 'LIBMBD_C6AU', 'IALL_IN_ONE', 'SMBJ', 'GAMMA_VDW', 'NSTORB', 'ML_LAFILT2', 'EFERMI', 'NRMM', 'VDW_ALPHA', 'ML_LMAX2', 'ML_IERR', 'LFOCKACE', 'ML_ISTART', 'BEXT', 'ML_IALGO_LINREG', 'ML_RCUT1', 'ML_NHYP', 'IVDW_NL', 'NHC_NS', 'CUTOFF_TYPE', 'LIBMBD_PARALLEL_MODE', 'IRC_DELTA0', 'LH5', 'ML_MRB2', 'XC', 'LIBMBD_R0AU', 'BSEHOLE', 'LIBXC2_Pn', 'PHON_G_CUTOFF', 'ML_IAFILT2', 'LALL_IN_ONE', 'FOCKCORR', 'PHON_NWRITE', 'ALDAX', 'ML_LHEAT', 'ML_RCOUPLE', 'AMGGAC', 'ML_IREG', 'STOP_ON', 'ML_NMDINT', 'NCORE_IN_IMAGE1', 'LCHARGH5', 'LIBMBD_ALPHA', 'ML_OUTBLOCK', 'ML_ICRITERIA', 'ESTOP', 'Construction:NBSEBLOCKO', 'LIBXC1_Pn', 'ML_WTSIF', 'CUTOFF_SIGMA', 'XC_C', 'IRC_DIRECTION', 'ML_LFAST', 'VDW_R0AU', 'NELMGW', 'Construction:NBSEBLOCKV', 'LIBMBD_TS_D', 'LPHON_READ_FORCE_CONSTANTS', 'ML_LSPARSDES', 'NMAXFOCKAE_and_LMAXFOCKAE', 'WRT_POTENTIAL', 'VDW_C6AU', 'IRC_MINSTEP', 'ML_MCONF', 'LIBXC2', 'ML_MHIS', 'ZAB_VDW', 'NHC_NCHAINS', 'LIBMBD_METHOD', 'KPOINTS_OPT_NKBATCH', 'LIBXC1', 'PHON_DIELECTRIC', 'VCAIMAGES', 'LFINITE_TEMPERATURE', 'ML_MCONF_NEW', 'Construction:LPARDH5', 'IRC_STOP', 'NBANDSEXACT', 'BSEPREC', 'LIBMBD_VDW_PARAMS_KIND', 'ML_OUTPUT_MODE', 'ML_MODE', 'Profiling', 'VCA', 'ML_SIGV0', 'ML_LMLFF', 'ML_SIGW0', 'LIBMBD_MBD_A', 'ML_MRB1', 'BSEELECTRON', 'LKPOINTS_OPT', 'VACPOTFLAT', 'NBANDS_WAVE', 'LIBMBD_N_OMEGA_GRID', 'LIBMBD_K_GRID_SHIFT', 'ML_SCLC_CTIFOR', 'ML_W1', 'ML_MB_MIN', 'ML_LERR', 'ML_NRANK_SPARSDES', 'LIBMBD_XC', 'Construction:EFOR', 'PHON_NEDOS', 'NOMEGA_DUMP', 'ML_LCOUPLE', 'LPHON_POLAR', 'ML_SION1', 'NTEMPER', 'NHC_NRESPA', 'NUM_WANN'
```

Might be good for me to write a script to automate such things, there're just too many tags to manually update...

Reply:
i assume the VASP docs cover both the latest and earlier versions. if the tags are no longer documented, they may have been deprecated a while ago. still probably makes sense for `incar_parameters.json` to be permissive, i.e. allow older tags that have been deprecated (though ideally with a warning to the user when deprecated tags are validated). maybe @esoteric-ephemera knows more

Reply:
The current [VASP wiki INCAR tags](https://www.vasp.at/wiki/index.php/Category:INCAR_tag) contain old and new tags, e.g. is IALGO a deprecated version of ALGO. But I can't say why there are some tags in the json that don't show up in the wiki.

Reply:
Thanks for the comment @QuantumChemist, looks like we got a small issue with the `get_incar_tags` method: https://github.com/materialsproject/pymatgen/blob/46d843c07b7dcfd1ab25ab41a01cfa60f5ce123d/src/pymatgen/io/vasp/help.py#L70-L76

Looks like the INCAR list page from VASP wiki has changed probably, there should be a total of 4 pages. I would update this now.

Reply:
Also I think the VASP wiki list is not complete, as e.g. LREAL is a tag that I use all the time but it doesn't show up on the list. It's in the wiki though https://www.vasp.at/wiki/index.php/LREAL.